### PR TITLE
Added three algo to recursion

### LINKED ISSUE #340: [General] Adding 3 more algo to the Recursion
### Issue Description
Adding 3 algo to Recursion
- Generate Paranthesis
- Count all subsequences with sum K
- Letter Combinations of a Phone number

### Additional Context
Solving and understanding these problems gives strong hold on recursion

---

**Would you like to work on this issue?**  
- [x] Yes  
- [ ] No  

@ajay-dhangar

### PR DESCRIPTION
**Pull request by govindumeesala**
### Description
Added 3 algo to Recursion
- Generate Paranthesis
- Count all subsequences with sum K
- Letter Combinations of a Phone number

Fixes #340 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

image

@ajay-dhangar check this out

### CODE REVIEW

File: docs/Recursion/Count-all-subsequences-with-sum-K.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Perfect Sum Problem | Count All Subsets with Sum k
+
+- Problem Statement: Given an array arr of size n of non-negative integers and an integer sum, the task is to count all subsets of the given array with a sum equal to the target sum. Due to potentially large outputs, the answer must be computed modulo 10^9+7.
+
+- Example:
+- Input:
+  n = 6, arr = [5, 2, 3, 10, 6, 8], sum = 10
+- Output: 3
+- Explanation: The subsets that sum to 10 are {5, 2, 3}, {2, 8}, and {10}.
```
Comment (ajay-dhangar) [inline docs/Recursion/Count-all-subsequences-with-sum-K.md:17]: replace 

```md
- Input:
  n = 6, arr = [5, 2, 3, 10, 6, 8], sum = 10
- Output: 3
- Explanation: The subsets that sum to 10 are {5, 2, 3}, {2, 8}, and {10}.
```

with

````md
```
- Input:
  n = 6, arr = [5, 2, 3, 10, 6, 8], sum = 10
- Output: 3
- Explanation: The subsets that sum to 10 are {5, 2, 3}, {2, 8}, and {10}.
```
````

File: docs/Recursion/generate-parantheses.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+tags: [generate-parentheses, backtracking, recursion, dsa]
+---
+
+## Generate Parentheses Problem | Return All Combinations of Well-Formed Parentheses
+
+- Problem Statement: Given n pairs of parentheses, the task is to generate all possible combinations of well-formed parentheses. A well-formed parentheses string must have an equal number of opening and closing brackets, with every opening bracket correctly paired with a closing one.
+
+- The solution involves recursively constructing valid strings by adding opening and closing brackets while maintaining balance.
+
+- Example: For n = 3, the output should be: ["((()))", "(()())", "(())()", "()(())", "()()()"].
```
Comment (ajay-dhangar) [inline docs/Recursion/generate-parantheses.md]: replace `For n = 3, the output should be: ["((()))", "(()())", "(())()", "()(())", "()()()"]` with ``` `For n = 3, the output should be: ["((()))", "(()())", "(())()", "()(())", "()()()"]` ```

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by govindumeesala** [comment]
> update code based on suggestions

@ajay-dhangar Updated changes . Please check once

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**